File: CSharp/Lesson4_LargestPalindrome.cs
using System;
using System.Linq;

namespace CSharp
{
    class Lesson4_LargestPalindrome
    {
        public void Get()
        {
            var threeDigitNumbers = Enumerable.Range(100, 900);

            threeDigitNumbers
                .SelectMany(a => threeDigitNumbers.Select(b => a * b))
                .Where(IsPalindrome)
               .Max();
        }

        private bool IsPalindrome(int arg)
        {
            return arg.ToString() == new string(arg.ToString().Reverse().ToArray());
        }
    }
}

File: FSharp/Lesson4_LargestPalindrome.fs
namespace FSharp

module Lesson4_LargestPalindrome = 
    open System
    let threeDigitNumbers = [100..999]

    let reverse (s:string) = String(s.ToCharArray() |> Array.rev)

    let isPalidrome a = a.ToString() = reverse (a.ToString())

    let max =  Seq.allPairs threeDigitNumbers threeDigitNumbers
               |> Seq.map (fun (a,b) -> a * b)
               |> Seq.filter isPalidrome
               |> Seq.max
    
        

